DependencyInjection/DoctrineExtension.php

```diff
@@ -254,6 +255,54 @@ protected function getConnectionOptions($connection)
     {
         $options = $connection;
 
+        $options['connection_override_options'] = [];
+
+        $connectionDefaults = [
+            'dbname' => null,
+            'host' => 'localhost',
+            'port' => null,
+            'user' => 'root',
+            'password' => null,
+        ];
+
+        $allowOverrideUrl = $options['override_url'] ?? false;
+
+        if (! $allowOverrideUrl && isset($options['url'])) {
+            Deprecation::trigger(
+                'doctrine/doctrineBundle',
+                'https://github.com/doctrine/DoctrineBundle/pull/1290',
+                'Not setting doctrine.dbal.override_url to true is deprecated. True is the only value that will be supported in doctrine-bundle 3.0.'
+            );
+        }
+
+        if ($allowOverrideUrl) {
+            foreach (array_keys($connectionDefaults) as $paramName) {
```

Comment:
Can be done simpler. Remember, PHP standard library is powerful.
```suggestion
            $options['connection_override_options'] = array_intersect_key($options, $connectionDefaults);
```

Reply:
good catch!